sabnzbd/newsunpack.py

```diff
@@ -1997,7 +1997,11 @@ def quick_check_set(setname: str, nzo: NzbObject) -> bool:
             # Do a simple filename based check
             if file == nzf.filename:
                 found = True
-                if (nzf.md5sum is not None) and nzf.md5sum == par2info.filehash:
+                if (
+                    (nzf.crc32sum is not None)
+                    and nzf.crc32sum == par2info.filehash
+                    and os.path.getsize(nzf.filepath) == par2info.filesize
```

Comment:
The file can be gone from the disk (due to a previous rename if obfosufated files appear in multiple par2 sets), so then this would crash. That's why the old code is wrapped in Try/except IOError: https://github.com/sabnzbd/sabnzbd/commit/2ac08dd0e6659b24e283ca2d3f9f62963d82a565#diff-4d3d76bbdc64028f5e5c1442647035ee0b14a7baef4a1f48d7caffb9e9992a6fL2120